SmartDeviceLink/SDLIAPSession.m

```diff
@@ -139,6 +273,51 @@ - (SDLStreamErrorHandler)streamErroredHandler {
     };
 }
 
+- (SDLStreamHasSpaceHandler)streamHasSpaceHandler {
+    __weak typeof(self) weakSelf = self;
+    
+    return ^(NSStream *stream) {
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        
+        if (strongSelf.isDataSession){
+            SessionSendHandler sendCompletion;
+            @synchronized (strongSelf) {
+                sendCompletion = strongSelf.sendBlock;
+            }
+            
+            if (sendCompletion != nil){
+                NSError *err = nil;
+                BOOL sendComplete = sendCompletion(&err);
+                if (err != nil){
+                    NSLog(@"ERROR!!! Output stream reported %@", err);
```

Comment:
Hi Joel, it looks like I had too little sleep yesterday and a bit too much caffeine. I pushed some in-progress changes to my develop branch fork and these commits wound up intermingled with the single focused change that I wanted to get reviewed. I think this feedback is great and you should know that I've corrected some of the issues, but the final design will use a FIFO for managing the in-flight send fragments rather than abusing a dispatch queue.